plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/guru/VmwareVmImplementer.java

```diff
@@ -139,6 +140,9 @@ VirtualMachineTO implement(VirtualMachineProfile vm, VirtualMachineTO to, long c
                     details.put(VmDetailConstants.NIC_ADAPTER, VirtualEthernetCardType.E1000.toString());
                 }
             }
+            if(StringUtils.isEmpty(details.get(VmDetailConstants.RAM_RESERVATION))){
+                details.put(VmDetailConstants.RAM_RESERVATION, "0.0");
```

Comment:
There are two occurrences of the "0.0" String.

What do you think of replacing these `"0.0"` with `NumberUtils.DOUBLE_ZERO`? The NumberUtils will return a `Double` object of `0.0d` and, therefore, it has a `toString()` that returns the String `"0.0"`.

Other option would be to create a constant with a self-describing name.

Reply:
Hi @GabrielBrascher, sorry for my late answer, I must have missed the notification.  I think there are no benefits regards reusable of this code, but I agree it's better to use parameter instead of strings. I'm prefer your first suggestion and will adapt  it.